pkg/kratos/service.go

```diff
@@ -0,0 +1,171 @@
+package kratos
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+	misc "github.com/canonical/identity-platform-login-ui/internal/misc/http"
+	"github.com/canonical/identity-platform-login-ui/internal/monitoring"
+	hClient "github.com/ory/hydra-client-go/v2"
+	kClient "github.com/ory/kratos-client-go"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	kratos KratosClientInterface
+	hydra  HydraClientInterface
+
+	tracer  trace.Tracer
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+// We override the type from the kratos sdk, as it does not
+// get marshalled correctly into json
```

Comment:
what is the problem with https://pkg.go.dev/github.com/ory/kratos-client-go#GenericError ?

Reply:
Maybe I am missing something. The struct in the sdk (https://pkg.go.dev/github.com/ory/kratos-client-go#ErrorGeneric) follows the convention described in the docs (https://www.ory.sh/docs/kratos/reference/api#tag/frontend/operation/updateLoginFlow), which is that the error should be like:
```json
{
  "error": {
    "error": {
      "code": 404,
      "debug": "SQL field \"foo\" is not a bool.",
      "details": {},
      "id": "string",
      "message": "The resource could not be found",
      "reason": "User with ID 1234 does not exist.",
      "request": "d7ef54b1-ec15-46e6-bccb-524b82c035e6",
      "status": "Not Found"
    }
  },
  "redirect_browser_to": "string"
}
```

BUT the response I get when running Kratos in docker(`v0.13.0`) locally, the response I get is:
```json
{
  "error": {
    "id": "browser_location_change_required",
    "code": 422,
    "status": "Unprocessable Entity",
    "reason": "In order to complete this flow please redirect the browser to: somewhere",
    "message": "browser location change required"
  },
  "redirect_browser_to": "somewhere"
}
```
The second one is missing one nesting level. I think that this happens because https://github.com/ory/kratos/blob/08fed36973274ef294491d00811bc867f1537d62/selfservice/flow/error.go#L167 is different from the one in the sdk. 

We could simply return a `RedirectBrowserTo` (without the `error` part) if we wanted, so that we don't have to define our own structure.

Reply:
great let's just add a permalink to this comment or add extra on the comment